network/manager.go

```diff
@@ -324,9 +309,54 @@ func (m *Manager) watchNetworks() {
 	}
 }
 
-func (m *Manager) Run() {
+func (m *Manager) addInitialNetworks(ctx context.Context) bool {
+	if m.isMultiNetwork() {
+		// Get list of existing networks
+		result, err := m.sm.WatchNetworks(ctx, nil)
+		if err != nil {
+			return false
+		}
+
+		for _, n := range result.Snapshot {
+			if m.isNetAllowed(n) {
+				m.networks[n] = NewNetwork(ctx, m.sm, m.bm, n, m.ipMasq)
+			}
+		}
+	} else {
+		if _, err := m.sm.GetNetworkConfig(ctx, ""); err != nil {
+			return false
+		}
+
+		m.networks[""] = NewNetwork(ctx, m.sm, m.bm, "", m.ipMasq)
+	}
+
+	return true
+}
+
+func (m *Manager) Run(ctx context.Context) {
 	wg := sync.WaitGroup{}
 
+	log.V(5).Info("Getting initial networks")
+
+	for i := 0; ; i++ {
+		// Try adding initial networks
+		if m.addInitialNetworks(ctx) {
+			log.V(5).Infof("Got %d initial networks", len(m.networks))
+			break
+		}
+
+		// Otherwise retry in a few seconds
+		select {
+		case <-ctx.Done():
+			return
+		case <-time.After(time.Second):
+		}
+
+		if i%5 == 0 {
```

Comment:
To reduce logspam, otherwise it would get printed every second.  I'd imagine this would often be a short-term log message though, until the etcd master or flannel master gets started up.